syntax = "proto2";
option optimize_for = LITE_RUNTIME;

import "common.proto";
package orwell.messages;

// Tell a player wants to join
// answered by: Welcome, Goodbye
message Hello {
	required string name = 1;
	optional bool ready = 2 [default = true];
	// zmq does not allow to retrieve the address, so we are going to trust
	// the clients to provide it (it is only for debugging purpose)
	optional string address = 3;
}

message Move {
	optional double left = 1;
	optional double right = 2;
}

message Fire {
	optional bool weapon1 = 1;
	optional bool weapon2 = 2;
}

// Notify the state of the inputs to control the robot
message Input {
	optional Move move = 1;
	optional Fire fire = 2;
}

// Ask for a pong message to mesure timings
message Ping {
	repeated Timing timing = 1;
}
